villanovachile/truthtally
# [Add] List Archives

### Description

closes #5
closes #6 

This PR finally brings the updates that completes the [Truth Tally - List Sharing project](https://github.com/orgs/villanovachile/projects/1). In order to bring the list archives to fruition, there needed to be a general site layout with traditional header, footer, navigation, etc. The Truth Tally app itself remains largely untouched, and doesn't flow with the site, so that is something that will need to be redesigned to fully integrate it in the site in a pretty way (I will need your input for that @ash1eygrace lol). 

### Additions
- List Archive pages for Ranked and Unranked lists with pagination, searching, and sorting.
- A new API endpoint `get_lists` which handles all requests for list archives from the frontend.
- Updated code for singleton connection to handle `get_lists`
- Contact page with implementation of ReCAPTCHA v3
- All list archives, searches, pagination, and sorting are handled via SSR

### Testing Instructions
When the staging deploy site is live below, be sure to test the following:

Ensure all testing is carried out both on desktop and mobile.

1. Test all functionality in the Truth Tally app itself, creating lists, sharing lists, ranked and unranked, listed and unlisted.
2. Ensure that the Ranked and Unranked archives display lists, and that searching, sorting, and pagination works.
3. Search for the unlisted and listed lists you created and shared display (listed) or do not display (for unlisted lists) on the page archives
4. Test all pages, including static footer pages, and test Contact page
5.  Test OG meta tags by entering a few pages, lists, archive pages at https://www.opengraph.xyz/

### Bugs/Issues/Suggestions in this PR

- [x] Search field on Unranked and Ranked lists pages is not reset when switching between Unranked and Ranked Lists pages. Reported by @villanovachile | fixed with 3d25289c2f00d8656dd549d75feceaafcad0fb45
- [x] Sort selection is not reset when switching between Unranked and Ranked lists pages. Reported by @villanovachile | fixed with 9d6b55e0cbd29819df01936c5ec24875a2af424b
- [x] Stray `;` on Create New List page. Reported by @villanovachile | fixed with adfea1d66751c830e5f5e3a45612a8a18997d104
- [x] Creating a list and sharing a list work fine, but I did notice that the share a list popup modal is cut off when the viewport is under ~700px. Reported by @ash1eygrace | fixed with adfea1d66751c830e5f5e3a45612a8a18997d104
- [x] On the [ranked](https://dev.truthtallyranker.com/lists/ranked) and [unranked](https://dev.truthtallyranker.com/lists/unranked) list pages it would be good for UX if the first options dropdown default was "All" vs. just "Title". Suggested by @ash1eygrace | fixed with 3d25289c2f00d8656dd549d75feceaafcad0fb45
- [x] Add frontend and backend validation/sanitization for contact form, unranked, and ranked lists pages. Reported by @villanovachile | fixed with 59aea222909e5c1f1621f9d07c6e4320521504bb
- [x]  Consistent use of buttons / boxes with rounded corners to match global style. Reported by @drardin | fixed with 3d25289c2f00d8656dd549d75feceaafcad0fb45 & 79fdbc05cba7b9417c6d174eff57afea810ad865
